django/core/management/commands/loaddata.py

```diff
@@ -223,6 +245,39 @@ def save_obj(self, obj):
             self.objs_with_deferred_fields.append(obj)
         return saved
 
+    def bulk_create_obj(self, objects):
+        """
+        split objects into groups with the same model, then use bulk_create to load
+        all instances in a single statement
+        """
+        objects_in_fixture = 0
+        loaded_objects_in_fixture = 0
+        group_model = None
+        instances = []
+
+        for obj in objects:
+            objects_in_fixture += 1
+            obj_model = obj.object.__class__
+            if not group_model:
+                group_model = obj_model
+            if obj_model == group_model:
+                instances.append(obj.object)
+                continue
+            group_model.objects.bulk_create(instances)
+            loaded_objects_in_fixture += len(instances)
+
+            group_model = obj_model
+            instances = [obj.object]
+
+        # repeat for last set of records
+        group_model.objects.bulk_create(instances)
+        loaded_objects_in_fixture += len(instances)
```

Comment:
Indeed that is one of the reasons. In theory, a group of articles could be separated by another object that the second group of articles is dependent on. By merely grouping and bulk-uploading, this could create a condition where a foreign key constraint is violated. Another aspect is that the current approach minimized memory consumption where possible.